numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1504,6 +1504,17 @@ arr_add_docstring(PyObject *NPY_UNUSED(dummy), PyObject *args)
 #include <emmintrin.h>
 #endif
 
+#if defined __ARM_NEON || defined __ARM_NEON__
```

Comment:
use the new definitions, replace it with `#ifdef NPY_HAVE_NEON`, and remove the NEON header.
NOTE: this will not enable `NEON` for armhf, unless the user builds NumPy with NEON or upper feature through `--cpu-baseline`
If it's necessary then you can use the new CPU dispatcher. if you're looking for an example, see [_umath_tests.dispatch.c](https://github.com/numpy/numpy/blob/master/numpy/core/src/umath/_umath_tests.dispatch.c) and [_umath_tests.c.src](https://github.com/numpy/numpy/blob/e8c78e09ab38b80b586108dfb4151cbe70c7a694/numpy/core/src/umath/_umath_tests.c.src#L579).

Reply:
Happy to use the new CPU dispatcher, but not sure how to apply to this case after studying two examples.

Reply:
The new CPU dispatcher way is kind of complicated, I will make another PR to improve. @seiko2plus what do you think?

Reply:
My bad, I'm going to speed up the doc progress, and I'm sure will find it too easy to use. In meanwhile you can leave it as-is, and we can integrate the code later with the CPU dispatcher during universal intrinsics replacement.